modules/kafka/src/main/scala/com/snowplowanalytics/snowplow/streams/kafka/sink/KafkaSink.scala

```diff
@@ -53,16 +53,23 @@ private[kafka] object KafkaSink {
     producer: KafkaProducer[String, Array[Byte]],
     ec: ExecutionContext
   ): Sink[F] =
-    Sink { batch =>
-      val f = Sync[F].delay {
-        val futures = batch.asIterable.map { e =>
-          val record = toProducerRecord(config, e)
-          producer.send(record)
-        }.toIndexedSeq
+    new Sink[F] {
+      def sink(batch: ListOfList[Sinkable]): F[Unit] = {
+        val f = Sync[F].delay {
+          val futures = batch.asIterable.map { e =>
+            val record = toProducerRecord(config, e)
+            producer.send(record)
+          }.toIndexedSeq
 
-        futures.foreach(_.get)
+          futures.foreach(_.get)
+        }
+        Async[F].evalOn(f, ec)
       }
-      Async[F].evalOn(f, ec)
+
+      def pingForHealth: F[Boolean] =
+        Sync[F].blocking {
+          producer.partitionsFor(config.topicName).asScala.nonEmpty
```

Comment:
There is no logs for `starting to check` / `checked successful`, it'd be nice to be consistent between destinations